remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala

```diff
@@ -160,6 +167,17 @@ class NettyTransportSettings(config: Config) {
     case _                 => getBoolean("tcp-reuse-addr")
   }
 
+  val ByteBufAllocator: ByteBufAllocator = getString("bytebuf-allocator-type") match {
+    case "pooled"        => PooledByteBufAllocator.DEFAULT
+    case "unpooled"      => UnpooledByteBufAllocator.DEFAULT
+    case "unpooled-heap" => new UnpooledByteBufAllocator(false)
+    case "adaptive"      => new AdaptiveByteBufAllocator()
+    case "adaptive-heap" => new AdaptiveByteBufAllocator(false)
+    case other => throw new IllegalArgumentException(
+        "Unknown 'bytebuf-allocator-type' [" + other + "]," +
```

Comment:
Could you add a def (package private) of form `private[netty] def deriveByteBufAllocator(allocatorType: String): ByteBufAllocator` and use this here?

The def can be tested in a unit test to see how it handles the 5 input types (and an invalid input too).

Reply:
I will do this when find time,  stress testing to 4 am, on bed now,  feel free to update this branch ,thanks